gmake fails on windows

<!--

    The template below contains optional suggestions. You
    can omit some information.

-->

## The Problem

After building and installing NQP, `gmake` fails:

```
c:\raku\rakudo>gmake
+++ Checking for moar NQP version
+++ Expanding   gen\moar\main-version.nqp
+++ Generating  gen\moar\rakudo.nqp
+++ Generating  gen\moar\Grammar.nqp
+++ Generating  gen\moar\World.nqp
+++ Generating  gen\moar\ModuleLoader.nqp
+++ Compiling   blib\Perl6\ModuleLoader.moarvm
+++ Generating  gen\moar\Ops.nqp
+++ Compiling   dynext\perl6_ops_moar.dll
+++ Compiling   blib\Perl6\Ops.moarvm
+++ Generating  gen\moar\Pod.nqp
+++ Compiling   blib\Perl6\Pod.moarvm
+++ Compiling   blib\Perl6\World.moarvm
+++ Generating  gen\moar\Actions.nqp
+++ Compiling   blib\Perl6\Actions.moarvm
+++ Compiling   blib\Perl6\Grammar.moarvm
+++ Generating  gen\moar\Compiler.nqp
+++ Generating  gen\moar\Optimizer.nqp
+++ Compiling   blib\Perl6\Optimizer.moarvm
+++ Compiling   blib\Perl6\Compiler.moarvm
+++ Generating  gen\moar\SysConfig.nqp
+++ Compiling   blib\Perl6\SysConfig.moarvm
+++ Compiling   rakudo.moarvm
+++ Generating  gen\moar\BOOTSTRAP\v6c.nqp
+++ Generating  gen\moar\Metamodel.nqp
+++ Compiling   blib\Perl6\Metamodel.moarvm
+++ Compiling   blib\Perl6\BOOTSTRAP\v6c.moarvm
+++ Compiling   blib\CORE.c.setting.moarvm
The following step can take a long time, please be patient.
Stage start      :   0.000
Stage parse      :  69.092
Stage syntaxcheck:   0.000
Stage ast        :   0.000
Stage optimize   :   7.257
Stage mast       :  14.755
Stage mbc        :   9.321
+++ Generating  gen\moar\BOOTSTRAP\v6d.nqp
+++ Compiling   blib\Perl6\BOOTSTRAP\v6d.moarvm
+++ Compiling   blib\CORE.d.setting.moarvm
The following step can take a long time, please be patient.
Stage start      :   0.000
Stage parse      :   0.409
Stage syntaxcheck:   0.000
Stage ast        :   0.000
Stage optimize   :   0.021
Stage mast       :   0.066
Stage mbc        :   0.028
+++ Generating  gen\moar\BOOTSTRAP\v6e.nqp
+++ Compiling   blib\Perl6\BOOTSTRAP\v6e.moarvm
+++ Compiling   blib\CORE.e.setting.moarvm
The following step can take a long time, please be patient.
Stage start      :   0.000
Stage parse      :   1.634
Stage syntaxcheck:   0.000
Stage ast        :   0.000
Stage optimize   :   0.150
Stage mast       :   0.328
Stage mbc        :   0.033
+++ Generating  rakudo-m.exe
+++ Generating  gen\moar\rakudo-debug.nqp
+++ Compiling   rakudo-debug.moarvm
+++ Compiling   inst-rakudo.exe
+++ Compiling   inst-rakudo-debug.exe
+++ Compiling   inst-rakudo-m.exe
+++ Compiling   inst-rakudo-debug-m.exe
+++ Compiling   inst-rakudow.exe
gcc: error: unrecognized command line option '--subsystem=windows'; did you mean '--machine=windows'?
gmake: *** [Makefile:1323: inst-rakudow.exe] Error 1
```

Line 1323 in the *Makefile* is:

```
	$(NOECHO)$(M_CC) -c -DSTATIC_EXEC_PATH="C:\\raku\\rakudo\\install\\bin\\$(M_RAKUDOW)" -DSTATIC_NQP_HOME="C:\\raku\\rakudo\\install\\share\\nqp" -DSTATIC_RAKUDO_HOME="C:\\raku\\rakudo\\install\\share\\perl6"  -DSUBSYSTEM_WINDOWS $(M_CFLAGS) $(M_MOAR_INC_PATHS) -o $(M_INST_RAKUDOW).o src\vm\moar\runner\main.c
```

## Environment
* Operating system: Windows 10
* Compiler version (`install\bin\moar.exe --version`): This is MoarVM version 2020.10-31-gb13f6a24c built with JIT support

--- a/src/vm/moar/runner/main.c
+++ b/src/vm/moar/runner/main.c
@@ -9,6 +9,7 @@
 #  include <sys/stat.h>
 #  include <process.h>
 #  include <shlwapi.h>
+#  include <io.h>
 #  if defined(_MSC_VER)
 #    define strtoll _strtoi64
 #  endif

--- a/src/vm/moar/runner/main.c
+++ b/src/vm/moar/runner/main.c
@@ -202,7 +202,7 @@ int set_std_handle_to_nul(FILE *file, int fd, BOOL read, int std_handle_type) {
 #endif
 
 #if defined(_WIN32) && defined(SUBSYSTEM_WINDOWS)
-int WinMain(HINSTANCE hInstance, HINSTANCE hPrevInstance, PSTR lpCmdLine, INT nCmdShow) {
+int wWinMain(HINSTANCE hInstance, HINSTANCE hPrevInstance, LPWSTR lpCmdLine, INT nCmdShow) {
     int argc;
     LPWSTR *wargv = CommandLineToArgvW(GetCommandLineW(), &argc);
     char **argv = MVM_UnicodeToUTF8_argv(argc, wargv);

--- a/tools/lib/NQP/Config/Rakudo.pm
+++ b/tools/lib/NQP/Config/Rakudo.pm
@@ -416,7 +416,7 @@ sub configure_moar_backend {
             $nqp_config->{subsystem_win_ld_flags} = '/subsystem:windows';
         }
         else {
-            $nqp_config->{subsystem_win_ld_flags} = '--subsystem=windows';
+            $nqp_config->{subsystem_win_ld_flags} = '-mwindows';
         }
 
         push @c_runner_libs, sprintf( $nqp_config->{'moar::ldusr'}, 'Shlwapi' );

